Detections/MultipleDataSources/Unauthorized_user_access_across_AWS_and_Azure.yaml

```diff
@@ -0,0 +1,147 @@
+            ( 
+            where AIPCall has "name"    
+            | extend APICallName = tostring(AIPCall.name), APICallCount = tostring(AIPCall["count"])
+            ) 
+        | distinct
+            AWSAlertTime,
+            ActivityType,
+            Severity,
+            AWSAlertId,
+            AWSAlertTitle,
+            AWSAlertDescription,
+            AWSAlertLink,
+            Arn,
+            AWSresourceType,
+            AWSNetworkEntity,
+            AWSAlertUserNameEntity,
+            InstanceType,
+            APICallName,
+            APICallCount      
+        );
+        // Define a variable 'Azure_sigin' to collect Azure portal Signing activity from SigninLogs Table
+        let Azure_sigin = materialize (SigninLogs
+            | where AppDisplayName == "Azure Portal"
+            | where isnotempty(OriginalRequestId)
+            | summarize 
+                totalAzureLoginEventId = dcount(OriginalRequestId), 
+                AzureFailedEventsCount = dcountif(OriginalRequestId, ResultType != 0), 
+                AzureSuccessfulEventsCount = dcountif(OriginalRequestId, ResultType == 0),
+                AzureSetOfFailedEvents = makeset(iff(ResultType != 0, OriginalRequestId, ""), 5), 
+                AzureSetOfSuccessfulEvents = makeset(iff(ResultType == 0, OriginalRequestId, ""), 5) 
+            by 
+                IPAddress, 
+                UserPrincipalName, 
+                bin(TimeGenerated, 1min), 
+                UserAgent,
+                ConditionalAccessStatus,
+                OperationName,
+                RiskDetail,
+                AuthenticationRequirement,
+                ClientAppUsed 
+            // Extracting the name and UPN suffix from UserPrincipalName
+            | extend
+                Name = tostring(split(UserPrincipalName, "@")[0]),
+                UPNSuffix = tostring(split(UserPrincipalName, "@")[1])
+        );
+        // Join 'AwsAlert' and 'Azure_sigin' on the AWS Network Entity and Azure IP Address
+        AwsAlert
+        | join kind=inner Azure_sigin on $left.AWSNetworkEntity == $right.IPAddress
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: IPAddress
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: Name
+      - identifier: UPNSuffix
+        columnName: UPNSuffix
+customDetails:
+  AWSAlertUserName: AWSAlertUserNameEntity
+  AWSArn: Arn
+  AWSresourceType: AWSresourceType
+  AWSInstanceType: InstanceType
+  AWSAPICallName: APICallName
+  AWSAPICallCount: APICallCount
+  AzureUserAgent: UserAgent
+  AzureUser: UserPrincipalName
+  AzureClientAppUsed: ClientAppUsed
+  AzConditionalAccess: ConditionalAccessStatus
+  AzureOperationName: OperationName
+  AzureRiskDetail: RiskDetail
+  AzAuthRequirement: AuthenticationRequirement
+  alertSeverity: Severity
+alertDetailsOverride:
+  alertDisplayNameFormat: "{{AWSAlertTitle}}"

```

Comment:
Suggest you update to something like:

"{{AWSNetworkEntity}} from {{AWSAlertTitle}} observed in Azure Singins with {{UserPrincipalName}}"

Reply:
Updated